llama.cpp

```diff
@@ -9788,12 +9795,15 @@ int llama_token_to_piece(const struct llama_model * model, llama_token token, ch
                 }
                 memcpy(buf, result.c_str(), result.length());
                 return result.length();
+            } else if (llama_is_user_defined_token(model->vocab, token)) {
+                std::string result = model->vocab.id_to_token[token].text;
+                if (length < (int) result.length()) {
+                    return -result.length();
+                }
+                memcpy(buf, result.c_str(), result.length());
+                return result.length();
             } else if (llama_is_control_token(model->vocab, token)) {
```

Comment:
Seems redundant, since we ignore the token regardless of it's a control one or not.

Reply:
Yes. But these are the token types I'd expect to be used. In an older version the code asserted that no other token type could occur, but I'm not sure how we should react (assertion, exception or ignorance).